<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BaseDatos</class>
 <widget class="QMainWindow" name="BaseDatos">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1120</width>
    <height>711</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <widget class="QScrollArea" name="scrollArea">
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>1100</width>
         <height>650</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QTabWidget" name="tablas">
          <property name="currentIndex">
           <number>10</number>
          </property>
          <widget class="QWidget" name="municipio">
           <attribute name="title">
            <string>municipio</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_13">
            <item row="0" column="0">
             <widget class="QTableView" name="tableView_municipio"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="poblacion">
           <attribute name="title">
            <string>poblacion</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_12">
            <item row="0" column="0">
             <widget class="QTableView" name="tableView_poblacion"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="nodo">
           <attribute name="title">
            <string>nodo</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_11">
            <item row="0" column="0">
             <widget class="QTableView" name="tableView_nodo"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="centro">
           <attribute name="title">
            <string>centro</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_10">
            <item row="0" column="0">
             <widget class="QTableView" name="tableView_centro"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="aplicaciones">
           <attribute name="title">
            <string>aplicaciones</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_9">
            <item row="0" column="0">
             <widget class="QTableView" name="tableView_aplicaciones"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="programa">
           <attribute name="title">
            <string>programa</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_8">
            <item row="0" column="0">
             <widget class="QTableView" name="tableView_programa"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="emailnodo">
           <attribute name="title">
            <string>emailnodo</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_7">
            <item row="0" column="0">
             <widget class="QTableView" name="tableView_emailnodo"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="telefononodo">
           <attribute name="title">
            <string>telefononodo</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_6">
            <item row="0" column="0">
             <widget class="QTableView" name="tableView_telefononodo"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="comarca">
           <attribute name="title">
            <string>comarca</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="0" column="0">
             <widget class="QTableView" name="tableView_comarca"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="mancomunidad">
           <attribute name="title">
            <string>mancomunidad</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="0">
             <widget class="QTableView" name="tableView_mancomunidad"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="Consulta">
           <attribute name="title">
            <string>Consulta SQL</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout">
            <item row="0" column="0">
             <widget class="QFrame" name="frame">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>40</height>
               </size>
              </property>
              <property name="frameShape">
               <enum>QFrame::StyledPanel</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <widget class="QLabel" name="label_consulta">
               <property name="geometry">
                <rect>
                 <x>10</x>
                 <y>15</y>
                 <width>75</width>
                 <height>14</height>
                </rect>
               </property>
               <property name="text">
                <string>Consulta Sql</string>
               </property>
              </widget>
              <widget class="QPushButton" name="pB_sql">
               <property name="geometry">
                <rect>
                 <x>1010</x>
                 <y>3</y>
                 <width>41</width>
                 <height>35</height>
                </rect>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset>
                 <normaloff>../../../usr/share/grx/asistencia/imagenes/iconos/sql.svg</normaloff>../../../usr/share/grx/asistencia/imagenes/iconos/sql.svg</iconset>
               </property>
               <property name="iconSize">
                <size>
                 <width>32</width>
                 <height>32</height>
                </size>
               </property>
              </widget>
              <widget class="QComboBox" name="comboBox_consulta">
               <property name="geometry">
                <rect>
                 <x>100</x>
                 <y>5</y>
                 <width>901</width>
                 <height>31</height>
                </rect>
               </property>
               <property name="editable">
                <bool>true</bool>
               </property>
               <item>
                <property name="text">
                 <string># Consulta de extensiones, teléfonos e email de ayuntamientos

SELECT n.nombre as Nodo,
    n.extension as Extensión,
    GROUP_CONCAT(DISTINCT tn.telefono ORDER BY tn.telefono SEPARATOR ' | ') as Teléfono,
    GROUP_CONCAT(DISTINCT en.email ORDER BY en.email SEPARATOR ' | ') as Email,
    n.ipLinea as IP
FROM asismun.nodo n 
INNER JOIN asismun.poblacion p ON n.idPoblacion = p.id
LEFT JOIN asismun.telefononodo tn ON n.id = tn.idNodo
LEFT JOIN asismun.emailnodo en ON n.id = en.idNodo
WHERE n.esAyuntamiento = 1 
GROUP BY n.id 
ORDER BY n.nombre;
</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string># Consulta de datos de programas, mostrando letras identificativas

SELECT n.nombre, pro.anio,
pro.portalWeb as 165_A,
pro.email as B,
pro.baseDatosJuridica as C,
pro.suscripcionDominio as D,
pro.perfilContratante as E,
pro.gestionMunicipal as 166_A,
pro.gestionEconomica as B,
pro.soporte as C,
pro.sedeElectronica as 167_A,
pro.epol as 168_A,
pro.epolMovil as B,
pro.siapol as C
FROM asismun.programa pro
INNER JOIN nodo n ON pro.idNodo = n.id
ORDER BY n.nombre ASC, pro.anio ASC;</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string># Consulta de nodos a los que les falta información de contacto (faltan teléfono o dirección)

SELECT n.id as idNodo,
    n.nombre as Nodo,
	p.nombre as Población,
    GROUP_CONCAT(DISTINCT tn.telefono ORDER BY tn.telefono SEPARATOR ' | ') as Teléfono,
    n.fax as Fax,
    GROUP_CONCAT(DISTINCT en.email ORDER BY en.email SEPARATOR ' | ') as Email,
    n.web as Web,
    n.tipoVia as TipoVia,
    n.nombreVia as NombreVia,
    n.numeroDireccion as NumeroVia,
    n.codigoPostal as CódigoPostal
FROM asismun.nodo n 
INNER JOIN asismun.poblacion p ON n.idPoblacion = p.id
LEFT JOIN asismun.telefononodo tn ON n.id = tn.idNodo
LEFT JOIN asismun.emailnodo en ON n.id = en.idNodo
WHERE n.tipoVia is null or tn.idNodo is null
GROUP BY n.id 
ORDER BY n.nombre;
</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string># Consulta de nodos de centros

SELECT c.nombre as Centro,
    GROUP_CONCAT(DISTINCT n.nombre ORDER BY n.nombre SEPARATOR ' | ') as Nodo
FROM asismun.centro c
LEFT JOIN nodo n ON c.id = n.idCentro
GROUP BY c.id
ORDER BY c.nombre;</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>
# Consulta de datos de contacto con ayuntamientos
# Datos necesarios para la página de Diputación

SELECT n.id as idNodo,
    case when ela.id is null
		then n.nombre else CONCAT(n.nombre, &quot; (ELA)&quot;)
	end as Nodo,
    case when ela.id is null
		then m.nombre else ela.nombre
	end as Municipio,
	p.nombre as Población,
    GROUP_CONCAT(DISTINCT tn.telefono ORDER BY tn.telefono SEPARATOR ' | ') as Teléfono,
    n.fax as Fax,
    GROUP_CONCAT(DISTINCT en.email ORDER BY en.email SEPARATOR ' | ') as Email,
    n.web as Web,
    n.tipoVia as TipoVia,
    n.nombreVia as NombreVia,
    n.numeroDireccion as NumeroVia,
    n.codigoPostal as CódigoPostal,
    case when (ela.id is null and m.urlEscudo is not null) or (ela.id is not null and ela.urlEscudo is not null)
		then &quot;SÍ&quot; else null
	end as escudo
FROM asismun.nodo n 
INNER JOIN asismun.poblacion p ON n.idPoblacion = p.id
INNER JOIN asismun.municipio m ON p.idMunicipio = m.id
LEFT JOIN asismun.municipio ela ON p.idEla = ela.id
LEFT JOIN asismun.telefononodo tn ON n.id = tn.idNodo
LEFT JOIN asismun.emailnodo en ON n.id = en.idNodo
WHERE n.ipLinea LIKE &quot;%10.100%&quot; 
GROUP BY n.id 
ORDER BY n.nombre;
</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string># Consulta de programas

SELECT n.nombre as Nodo, p.*
FROM asismun.programa p
INNER JOIN nodo n ON p.idNodo = n.id
ORDER BY n.nombre;</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string># Consulta de poblaciones por municipios

SELECT m.nombre as Municipio, 
    GROUP_CONCAT(DISTINCT p.nombre ORDER BY p.nombre SEPARATOR ' | ') as Población
FROM municipio m
INNER JOIN poblacion p ON m.id = p.idMunicipio
GROUP BY m.id
ORDER BY m.nombre;</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string># Consulta de nodos por poblaciones

SELECT p.nombre as Población, 
    GROUP_CONCAT(DISTINCT n.nombre ORDER BY n.nombre SEPARATOR ' | ') as Nodo
FROM asismun.poblacion p
INNER JOIN nodo n ON p.id = n.idPoblacion
GROUP BY p.id
ORDER BY p.nombre;
</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string># Consulta de total de routers por tipo

SELECT n.equipamientoLinea as Router, 
	count(*) as Total,
    GROUP_CONCAT(DISTINCT n.caudalLinea ORDER BY n.caudalLinea SEPARATOR ' | ') as Caudales
FROM nodo n
GROUP BY n.equipamientoLinea
ORDER BY n.equipamientoLinea;</string>
                </property>
               </item>
              </widget>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QTableView" name="tableView_consulta"/>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1120</width>
     <height>19</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
